python/cuml/preprocessing/model_selection.py

```diff
@@ -225,17 +229,25 @@ def train_test_split(
         if train_size is None:
             train_size = X.shape[0] - test_size
 
+    if hasattr(X, "__cuda_array_interface__"):
+        x_order = _strides_to_order(X.__cuda_array_interface__['strides'],
+                                    cp.dtype(X.dtype))
+
+    if hasattr(y, "__cuda_array_interface__"):
+        y_order = _strides_to_order(y.__cuda_array_interface__['strides'],
+                                    cp.dtype(y.dtype))
+
     if cuda.is_cuda_array(X) or isinstance(X, cp.sparse.csr_matrix):
-        X_train = X[0:train_size]
-        y_train = y[0:train_size]
+        X_train = cp.array(X[0:train_size], order=x_order)
+        y_train = cp.array(y[0:train_size], order=y_order)
     elif isinstance(X, cudf.DataFrame):
         X_train = X.iloc[0:train_size]
         y_train = y.iloc[0:train_size]
 
-    if cuda.is_cuda_array(y) or isinstance(X, cp.sparse.csr_matrix):
-        X_test = X[-1 * test_size:]
-        y_test = y[-1 * test_size:]
-    elif isinstance(y, cudf.DataFrame):
+    if cuda.is_cuda_array(X) or isinstance(X, cp.sparse.csr_matrix):
```

Comment:
why is_cuda_array here and is_cuda_ndarray above in x_numba assignment?

Reply:
is_cuda_array returns True for any object that has cuda_array_interface. To be honest, I think we can just change this to `hasattr(X, "__cuda_array_interface__")` all around the codebase